src/Services/Shop/Shopify.php

```diff
@@ -208,6 +209,41 @@ public function customer($id): Customer
         return $this->mapCustomer($customer);
     }
 
+    /**
+     * @param array $data
+     * @return array
+     */
+    public function storeDiscount(array $data): array
+    {
+        $required_fields = ['title', 'value_type', 'value', 'starts_at'];
+
+        foreach ($required_fields as $field) {
+            if (!Arr::has($data['price_rule'], $field)) {
+                throw new Exception('price_rule.' . $field . ' is a required field');
+            }
+        }
+
+        $priceRule = $this->request('post', 'price_rules', $data);
+
+        $discountCodes = $this->request('post', 'price_rules/' . Arr::get($priceRule, 'price_rule.id') . '/batch', $data);
+
+        return [
+            'price_rule' => $priceRule,
+            'discount_codes' => $discountCodes
+        ];
+    }
+
+    /**
+     * Retrieves a DiscountCode
+     * @param string $code
+     * @return array|void
+     */
+    public function showDiscount(string $code): array
+    {
+        return $this->request('post', 'price_rules/' . $code . '/discount_codes/' . $code .  '.json');
```

Comment:
Dit is een get request

Reply:
en hij klopt niet price rules zit niet in de request dit is de opbouw:
GET /admin/api/2020-10/discount_codes/lookup.json?code=SUMMERSALE10OFF